packages/babel-generator/src/printer.ts

```diff
@@ -209,7 +218,25 @@ class Printer {
     }
 
     this._maybeAddAuxComment();
-    this._append(str);
+    this._append(str, maybeNewline);
+  }
+
+  tokenChar(char: number): void {
+    // space is mandatory to avoid outputting <!--
+    // http://javascript.spec.whatwg.org/#comment-syntax
+    const lastChar = this.getLastChar();
+    if (
+      // Need spaces for operators of the same kind to avoid: `a+++b`
+      (char === charCodes.plusSign && lastChar === charCodes.plusSign) ||
+      (char === charCodes.dash && lastChar === charCodes.dash) ||
+      // Needs spaces to avoid changing '34' to '34.', which would still be a valid number.
+      (char === charCodes.dot && this._endsWithInteger)
+    ) {
+      this._space();
+    }
```

Comment:
Q: Does the generator know if we are printing a module or a script? In modules we don't need the space, se we could skip these checks.

Reply:
I'm not sure about this, I'm not too familiar with logic yet...

Reply:
Ok let's keep them for now, we'll think about it in the future :+1: